pkg/distributor/distributor.go

```diff
@@ -765,6 +772,18 @@ func (d *Distributor) Push(ctx context.Context, req *cortexpb.WriteRequest) (*co
 	d.receivedExemplars.WithLabelValues(userID).Add(float64(validatedExemplars))
 	d.receivedMetadata.WithLabelValues(userID).Add(float64(len(validatedMetadata)))
 
+	nhRateLimited := false
+	if limits.NativeHistogramsIngestionRate != math.MaxFloat64 {
+		nhRateLimited = !d.nativeHistogramsIngestionRateLimiter.AllowN(time.Now(), userID, validatedHistogramSamples)
+	}
+
+	if nhRateLimited {
+		d.validateMetrics.DiscardedSamples.WithLabelValues(validation.NativeHistogramsRateLimited, userID).Add(float64(validatedHistogramSamples))
```

Comment:
If native histogram is ratelimited, do we still include validated histogram samples on Line 794? Maybe you should reset validatedHistogramSamples to 0 as we are not going to ingest histogram samples

```
totalSamples := validatedFloatSamples + validatedHistogramSamples
```

Reply:
Let's also add a log, to provide more information in addition to the metric

Reply:
Thanks updated. I have added warn log.